### PR TITLE
Move tar function to a new package

### LINKED ISSUE #2428: Move tar function to a new package
**Is your feature request related to a problem? Please describe.**
Move tar function to a new package and make tar functions exported so that other package can reuse the code.

**Describe the solution you'd like**

**Describe alternatives (optional)**

**Additional Notes**

### PR DESCRIPTION
**Pull request by lhw362950217**
Fixes #2428

### CODE REVIEW

File: pkg/tar/tar_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
```
Comment (typhoonzero) [inline pkg/tar/tar_test.go:1]: Should remove original `tar_test.go`

Comment (lhw362950217) [reply pkg/tar/tar_test.go:1]: done

**Review by typhoonzero — DISMISSED** [review]

File: pkg/modelzooserver/modelzooserver_test.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func TestModelZooServer(t *testing.T) {
 		err = os.Chdir(dir)
 		a.NoError(err)
 
-		err = tar.TarGzDir(".", "modelrepo.tar.gz")
+		err = tar.GzDir(".", "modelrepo.tar.gz")
```
Comment (typhoonzero) [inline pkg/modelzooserver/modelzooserver_test.go]: Should change both function names: `ZipDir`, `UnzipDir`

Comment (lhw362950217) [reply pkg/modelzooserver/modelzooserver_test.go]: done

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**